backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/scheduling/utils/VdsCpuUnitPinningHelper.java

```diff
@@ -127,8 +138,9 @@ public List<VdsCpuUnit> updatePhysicalCpuAllocations(VM vm, Map<Guid, List<VdsCp
             return cpusToBeAllocated;
         }
 
-        filterSocketsWithInsufficientMemoryForNumaNode(cpuTopology, vm, hostId);
```

Comment:
given the changes i did now in https://github.com/oVirt/ovirt-engine/pull/380, we run this allocation: `VMs * hosts * 2 (once pinning policy, once CPU policy) + VMs(schedule)` .

but: in terms of logical thinking - i think the current implementation make sense, also it saves us big scheduler change.
in terms of code and performance - we should do all the filtering once to reduce the `*2` part.

Reply:
I think it won't be possible to merge `CpuUnitPolicy` and `CpuPinningPolicyUnit` into one unit after all. It would be possible if we could rely that all of the hosts do report cpuTopology and we can do manual pinning on the cpuToplogy directly but that might not be true for older hosts. For those, we still need to count with the currently reported online cpus and it makes sense to have separated in `CpuPinningPolicyUnit`.